app/modules/listings/components/listing_element_comment.svelte

```diff
@@ -0,0 +1,121 @@
+<script lang="ts">
+  import autosize from 'autosize'
+  import { userContent } from '#app/lib/handlebars_helpers/user_content'
+  import { icon } from '#app/lib/icons'
+  import { getActionKey } from '#app/lib/key_events'
+  import Spinner from '#components/spinner.svelte'
+  import { updateElement } from '#listings/lib/listings'
+  import { I18n, i18n } from '#user/lib/i18n'
+
+  export let isCreatorMainUser, flash, element
+
+  let isEditMode, validating
+  let newComment = element.comment
+
+  async function validate () {
+    validating = _updateElement()
+    await validating
+    toggleEditMode()
+  }
+
+  function onKeyDown (e) {
+    const key = getActionKey(e)
+    if (key === 'esc') isEditMode = false
+    else if (e.ctrlKey && key === 'enter') validate()
+    e.stopPropagation()
+  }
+
+  function toggleEditMode () {
+    isEditMode = !isEditMode
+    flash = null
+  }
+
+  $: comment = element.comment
+  async function _updateElement () {
+    flash = null
+    try {
```

Comment:
[non-blocking]
That would be the perfect place to insert the spam filter introduced by 22a77a8c9
```suggestion
    try {
      await checkSpamContent(newComment)
```

Reply:
Yes, lets add it once its rebased or after merge to not rewrite all commits ids in a rebase

Reply:
since it got rebased, done in adad16576